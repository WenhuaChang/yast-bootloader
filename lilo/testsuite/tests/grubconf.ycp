/**
 * File:
 *  lilo2Grub.ycp
 *
 * Module:
 *  Bootloader configurator
 *
 * Summary:
 *  Translating device names from UNIX to grub
 *
 * Authors:
 *  Jiri Srain <jsrain@suse.cz>
 *
 * $Id$
 *
 */

{
    include "testsuite.ycp";
    map IREAD = $["target" : $["string" : "root (hd0,0)\ninstall /grub/stage1 d (hd0) /grub/stage2 0x8000 (hd0,0)/grub/menu.lst\nquit"],
		  "etc" : $["fstab" : []],
		  "probe" : $["disk" : [], "boot_arch" : "grub", "architecture" : "i386", "has_apm" : true, "has_pcmcia" : false, "has_smp" : false, "system" : []]];
    TESTSUITE_INIT ([IREAD, $[], $[]], 0);
    include "lilo/misc.ycp";
    include "lilo/routines.ycp";

    TEST (``(parseGrubConf ()), [$["target" : $["string" : "root (hd0,0)
install /grub/stage1 d (hd0) /grub/stage2 0x8000 (hd0,0)/grub/menu.lst
quit
"]], $[], $[]], 0);
    TEST (``(parseGrubConf ()), [$["target" : $["string" : "root (hd0,0)
install --stage2=/boot/grub/stage2 /grub/stage1 d (hd0) /grub/stage2 0x8000 (hd0,0)/grub/menu.lst
quit
"]], $[], $[]], 0);

    TEST (``(splitstring (getGrubConf ($["addr":"0x8000", "device":"(hd0)", "discswitch":true, "menu":"(hd0,0)/grub/menu.lst", "root":"(hd0,0)", "s2prefix":"--stage2=/boot/grub/stage2", "stage1":"/grub/stage1", "stage2":"/grub/stage2"]), "\n")), [$[], $[], $[]], 0);


}

