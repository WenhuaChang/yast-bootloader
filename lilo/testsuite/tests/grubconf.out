Read	.probe.architecture "i386"
Read	.probe.has_apm true
Read	.probe.has_pcmcia false
Read	.probe.has_smp false
Read	.probe.system []
Read	.etc.fstab []
Read	.target.string "/etc/grub.conf" "root (hd0,0)\ninstall /grub/stage1 d (hd0) /grub/stage2 0x8000 (hd0,0)/grub/menu.lst\nquit"
Read	.probe.disk []
Read	.probe.boot_arch "grub"
Read	.target.size "/etc/sysconfig/bootloader" 0
Read	.target.string "/etc/grub.conf" "root (hd0,0)\ninstall /grub/stage1 d (hd0) /grub/stage2 0x8000 (hd0,0)/grub/menu.lst\nquit\n"
Return	$["addr":"0x8000", "device":"(hd0)", "discswitch":true, "menu":"(hd0,0)/grub/menu.lst", "root":"(hd0,0)", "stage1":"/grub/stage1", "stage2":"/grub/stage2"]
Read	.target.string "/etc/grub.conf" "root (hd0,0)\ninstall --stage2=/boot/grub/stage2 /grub/stage1 d (hd0) /grub/stage2 0x8000 (hd0,0)/grub/menu.lst\nquit\n"
Return	$["addr":"0x8000", "device":"(hd0)", "discswitch":true, "menu":"(hd0,0)/grub/menu.lst", "root":"(hd0,0)", "s2prefix":"--stage2=/boot/grub/stage2", "stage1":"/grub/stage1", "stage2":"/grub/stage2"]
Return	["root (hd0,0)", "install --stage2=/boot/grub/stage2 /grub/stage1 d (hd0) /grub/stage2 0x8000 (hd0,0)/grub/menu.lst", "quit", ""]
